### PR TITLE
Stricter Eslint Part 2

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

IMHO there is not much sense in having rules as "warning", either they are errors (so we fix them) or we don't care, so we disable them.

Since we are not using next/image (at the moment) I disabled that rule. If we want to use it at some point, we just need to enable that rule and errors will start showing.

For the other rules I raised the warning to error and fix every problem in the code.

This PR aims to improve quality of frontend code.

### CODE REVIEW

File: frontend/.eslintrc [hunk 1]
```diff
@@ -7,11 +7,12 @@
...
   ],
   "rules": {
-    "@typescript-eslint/no-unused-vars": "warn",
-    "@typescript-eslint/no-explicit-any": "warn",
-    "@typescript-eslint/explicit-module-boundary-types": "off", //This one is to allow implicit type casting in functional components, otherwise we should use JSX.Element.
-    "react/display-name": "off", //This rule is to avoid the warning of nexts dynamic components with displayName
-    "@next/next/no-img-element": "warn", // Changed to warn since next Image component is bugged without sharp. From Next 11.2 on it will be fixed.
+    "@typescript-eslint/no-unused-vars": "error",
+    "@typescript-eslint/no-explicit-any": "error",
+    "react-hooks/exhaustive-deps": "error",
```
Comment (jordisala1991) [inline frontend/.eslintrc:12]: This one was reported as a warning, again, better to show as an error and fix the problems (another option is to disable the rule, but I think it provides useful tips to refactor hooks)

File: frontend/src/components/App/PreFishbowl/PreFishbowlParticipants.tsx [hunk 2]
```diff
@@ -68,6 +54,20 @@ const PreFishbowlParticipants = () => {
   };
 
   useEffect(() => {
+    const pingParticipant = () => {
```
Comment (aerrasti) [inline frontend/src/components/App/PreFishbowl/PreFishbowlParticipants.tsx:57]: Why this function need to be inside useEffect?

Comment (jordisala1991) [reply frontend/src/components/App/PreFishbowl/PreFishbowlParticipants.tsx:57]: because it is not used outside the useEffect

Comment (aerrasti) [reply frontend/src/components/App/PreFishbowl/PreFishbowlParticipants.tsx:57]: the eslint sends the error if is not used outside the useEffect?

Comment (jordisala1991) [reply frontend/src/components/App/PreFishbowl/PreFishbowlParticipants.tsx:57]: yes. If you use the function on multiple useEffect you can use `useCallback` , you have multiple examples on the PR.

**Comment by jordisala1991** [comment]
This PR became only the enable of: "react-hooks/exhaustive-deps": "error",

**Pull request closed without merging**